http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/impl/ApacheHttpRequestFactory.java

```diff
@@ -71,7 +71,8 @@ public HttpRequestBase create(final HttpExecuteRequest request, final ApacheHttp
      */
     private String sanitizeUri(URI uri) {
         String newPath = uri.getPath().replace("//", "/%2F");
-        return uri.toString().replace(uri.getPath(), newPath);
+        String baseUri = uri.toString();
```

Comment:
Query : 
`String newPath = uri.getPath().replace("//", "/%2F");`
Here we are handling only. "/" what if we havae other special characters ? 
Could you please test it for "@" or try the example given below
```

https://docs.oracle.com/javase/7/docs/api/java/net/URLEncoder.html
For example using UTF-8 as the encoding scheme the string "The string ü@foo-bar" would get converted to "The+string+%C3%BC%40foo-bar" because in UTF-8 the character ü is encoded as two bytes C3 (hex) and BC (hex), and the character @ is encoded as one byte 40 (hex).
```

URLEncoder.encode() might handle for all special characters .. What do you say ?

Reply:
The uri has already been encoded in https://github.com/aws/aws-sdk-java-v2/blob/master/http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpRequest.java#L108 at this point :)

I will create another PR for the fix

Reply:
got it..Thank you.